### PR TITLE
This is the initial implementation for interfacing with devicedb

### PR DESCRIPTION
**Pull request by SenRamakri**
This is the initial implementation for interfacing with devicedb. This can be now be used for listening to network config changes if stored in devicedb. New tests have been added in config_test.go. A new docker container running devicedb has been added to .travis.yml for testing.

### CODE REVIEW

**Comment by SenRamakri** [comment]
@cmonr - This can be reviewed/merged now, I have all the tests/docker container added and running now.

**Comment by cmonr** [comment]
Ugh. Will need to look at this outside of GitHub.

There's no good way to exclude reviewing files in `vendor/`.

**Comment by cmonr** [comment]
Checking updated branch settings.

File: .travis.yml [hunk 1]
```diff
@@ -29,11 +35,19 @@ script:
   # Build maestro
   - ./build.sh
 
+  # Run devicedb in docker
+  - |
+    if [[ "$TRAVIS_OS_NAME" == "linux" ]]; then
+      docker run --rm -d -p 127.0.0.1:9090:9090/tcp senramakri/test_images:devicedb_nocluster
+      sleep 10
```
Comment (cmonr) [inline .travis.yml]: I take it this line waits for devicedb to be brought up? Is there no cleaner /definitive way to detect when the service is ready?

File: .travis.yml [hunk 2]
```diff
@@ -29,11 +35,19 @@ script:
...
+    fi
+
   # Run unit tests
   - cd "../../${upstream}"
   - DEBUG=1 DEBUG2=1 ./build.sh
   - ( cd sysstats && LD_LIBRARY_PATH=../../maestro/vendor/github.com/armPelionEdge/greasego/deps/lib/ go test -v )
   - ( cd maestroConfig && LD_LIBRARY_PATH=../../maestro/vendor/github.com/armPelionEdge/greasego/deps/lib/ go test -v )
+  - ( cd vendor/github.com/armPelionEdge/maestroSpecs && go test -v )
```
Comment (cmonr) [inline .travis.yml]: Making sure, does this not need the LD_LIBRARY_PATH env var set prior to running?

File: maestroConfig/config_test.go [hunk 3]
```diff
@@ -83,13 +87,16 @@ func (cfgHook ConfigChangeHook) SawChange(configgroup string, fieldchanged strin
...
 	return true;
 }
 
-
+//Global wait group
+var wg sync.WaitGroup
+var sawChangeCount int = 0
 func TestConfigMonitorSimple(t *testing.T) {
-	var devicedbUri string = "https://WWRL000000:9090" //The URI of the relay's local DeviceDB instance
+	var devicedbUri string = "https://WWRL000000:9090" //The URI of the relay's local DeviceDB instan*client.e
```
Comment (cmonr) [inline maestroConfig/config_test.go]: Is `instan*client.e` a typo?

Comment (SenRamakri) [reply maestroConfig/config_test.go]: Yup, it is a typo.

File: .travis.yml (hunk 2 above)
Comment (SenRamakri) [reply .travis.yml]: Nope, that doesnt need the path ad it doesnt depend on the lib.

File: .travis.yml (hunk 1 above)
Comment (SenRamakri) [reply .travis.yml]: Usually it doesnt take that long, if it didnt start the tests will fail and will clearly say it cant connect. So we can detect that easily from logs. Anyway, let me know if we need a definite way.

**Comment by SenRamakri** [comment]
@cmonr - let me push a commit to address the comments.

File: maestroConfig/config_test.go (hunk 3 above)
Comment (SenRamakri) [reply maestroConfig/config_test.go]: @cmonr - This is fixed now.

**Pull request merged**